### PR TITLE
Revert "Switch to libsamplerate's callback API in `Sample`"

### PR DESCRIPTION
**Pull request by sakertooth**
Reverts LMMS/lmms#7361

### CODE REVIEW

**Comment by sakertooth** [comment]
Sorry, but inviting @AW1534 again to try the build out for this PR [here](https://lmms.io/download/pull-request/7410).

**Comment by AW1534** [comment]
@sakertooth This appears to have fixed the issue

**Comment by sakertooth** [comment]
It seems like the investigation continues. The team plans to do some more debugging on this, so I will close this for now. If all else fails, this PR can be reopened as a fallback solution.

**Review by Rossmaxx — APPROVED** [review]
Ok then.

File: include/Sample.h [hunk 1]
```diff
@@ -25,44 +25,55 @@
...
 
 namespace lmms {
 class LMMS_EXPORT Sample
 {
 public:
+	// values for buffer margins, used for various libsamplerate interpolation modes
+	// the array positions correspond to the converter_type parameter values in libsamplerate
+	// if there appears problems with playback on some interpolation mode, then the value for that mode
+	// may need to be higher - conversely, to optimize, some may work with lower values
+	static constexpr auto s_interpolationMargins = std::array<int, 5>{64, 64, 64, 4, 4};
```
Comment (Rossmaxx) [inline include/Sample.h:44]: Btw will it make sense to push this back to gigplayer ?

Comment (sakertooth) [reply include/Sample.h:44]: Maybe, but if we remove libsamplerate and use our own interpolation functions, there's no need.

**Pull request merged**